File: frontend/src/types/forecast.ts
export interface ForecastPoint {
  start: string;
  end: string;
  moer_lbs_per_mwh: number;
  health_score: number;
}

export interface ForecastResponse {
  region: string;
  generated_at: string;
  interval_minutes: number;
  points: ForecastPoint[];
  source: "live" | "cache" | "fixture";
}

export interface SignalIndexRegion {
  ba: string;
  index: number;
}

export interface SignalIndexResponse {
  generated_at: string;
  regions: SignalIndexRegion[];
}

export interface WeatherHour {
  time: string;
  temperature_c: number;
}

export interface WeatherResponse {
  latitude: number;
  longitude: number;
  hourly: WeatherHour[];
  source: string;
}

File: backend/app/models/forecast.py
from pydantic import BaseModel
from typing import Literal


class ForecastPoint(BaseModel):
    start: str
    end: str
    moer_lbs_per_mwh: float
    health_score: float


class ForecastResponse(BaseModel):
    region: str
    generated_at: str
    interval_minutes: int = 5
    points: list[ForecastPoint]
    source: Literal["live", "cache", "fixture"]


class SignalIndexRegion(BaseModel):
    ba: str
    index: int


class SignalIndexResponse(BaseModel):
    generated_at: str
    regions: list[SignalIndexRegion]
